### DISCUSSION TITLE
Page rect position convert !

### DISCUSSION THREAD

**Discussion by heyhumor** [pymupdf/PyMuPDF#1590] [Q&A] [answered]
If i creat a new portrait page, the start position at top left. then i set rotation with 90°, page turned landscape and the start position at top right. OK, how do i change position start at top left in landscape. This is not a bug, Thanks for you advice!

**Comment by JorjMcKie** [comment, marked as answer]
So you used `page.set_rotation(90)`.
The next steps all depend on what you want to do with the page.
1. All viewers will show the page rotated - just as if the former bottom-left now is top-left.
2. If you you however extract things from the page (text, drawings, images), the accompanying coordinates are all **_as if the page were not rotated._**
3. If you insert stuff on that page, the coordinates you provide **_are also assumed to be unrotated!_** Internally, the page will be derotated before insertion and then rotated back.
4. If you wish to insert respecting rotation, you must **_convert coordinates first_**, using `page.derotation_matrix`. If inserting e.g. text, also remember to insert it rotated (if that is what you actually want). Similarly images or drawings. The following inserts text at point (100, 100) in new, rotated coordinates. Text shall be oriented as if it were an original landscape page:

```python
>>> doc=fitz.open("1text.pdf")
>>> page=doc[0]
>>> page.set_rotation(90)
>>> p=fitz.Point(100,100) * page.derotation_matrix  # converted point coordinates
>>> page.insert_text(p, "Hallo", rotate=90, color=(1,0,0))
1
>>> doc.save("x.pdf")
>>> 
```
Result:
![grafik](https://user-images.githubusercontent.com/8290722/152767547-d661d611-4afb-4aca-af7c-490b0fb77579.png)
